forch/faucetizer.py

```diff
@@ -208,12 +208,9 @@ def _faucetize(self):
                 if self._has_acl(acl_name):
                     port_cfg['acls_in'] = [acl_name]
                 else:
-                    port_cfg['acls_in'] = []
                     LOGGER.error('No ACL defined for role %s', device_behavior.role)
-            else:
-                port_cfg['acls_in'] = []
             if self._config.tail_acl:
-                port_cfg['acls_in'].append(self._config.tail_acl)
+                port_cfg.setdefault('acls_in', []).append(self._config.tail_acl)
```

Comment:
try to be consistent in the code in the treatment of this setting, so, e.g., above you should use setdefault(() as well rather than setting = [acl_name] -- the reason is consistency and more straightforward flow so I don't need to worry abotut "has the variable been previously initialized".

Reply:
I think the problem is that in line 178, faucetizer already added tail_acl to all the host ports. If we use setdefault in line 209, it will append the role acl to the existing list ([tail_acl]), which will finally result in [tail_acl, role_acl, tail_acl].  I think maybe it is correct to explicitly overwrite the port config in line 209, since we have a device behavior for that port and hence that port needs to reconfigured based on the device behavior.

Reply:
the right solution is to add the tail_acl at the end of the flow, not the beginning... overwriting likely isn't the way to go.